deploy-openshift/kube-ops-view-oauth.yaml

```diff
@@ -0,0 +1,194 @@
+            timeoutSeconds: 1
+          resources:
+            limits:
+              cpu: 200m
+              memory: 100Mi
+            requests:
+              cpu: 50m
+              memory: 50Mi
+          volumeMounts:
+          - mountPath: /kube_ops_view_oauth
+            name: kube-ops-view-oauth
+            readOnly: true
+        restartPolicy: Always
+        serviceAccount: kube-ops-view
+        volumes:
+        - name: kube-ops-view-oauth
+          secret:
+            secretName: kube-ops-view-oauth
+    triggers:
+    - type: ConfigChange
+- kind: DeploymentConfig
+  apiVersion: v1
+  metadata:
+    labels:
+      app: kube-ops-view-redis
+    name: kube-ops-view-redis
+  spec:
+    replicas: 1
+    selector:
+      app: kube-ops-view-redis
+      deploymentconfig: kube-ops-view-redis
+    strategy:
+      activeDeadlineSeconds: 21600
+      rollingParams:
+        intervalSeconds: 1
+        maxSurge: 25%
+        maxUnavailable: 25%
+        timeoutSeconds: 600
+        updatePeriodSeconds: 1
+      type: Rolling
+    template:
+      metadata:
+        labels:
+          app: kube-ops-view-redis
+          deploymentconfig: kube-ops-view-redis
+      spec:
+        serviceAccount: kube-ops-view
+        containers:
+        - name: redis
+          image: redis:3.2-alpine
+          imagePullPolicy: Always
+          ports:
+          - containerPort: 6379
+            protocol: TCP
+          readinessProbe:
+            tcpSocket:
+              port: 6379
+          resources:
+            limits:
+              cpu: 200m
+              memory: 100Mi
+            requests:
+              cpu: 50m
+              memory: 50Mi
+    triggers:
+    - type: ConfigChange
+- kind: Route
+  apiVersion: v1
+  metadata:
+    labels:
+      app: kube-ops-view
+    name: kube-ops-view
+  spec:
+    host: "${KUBE_OPS_VIEW_APP_DOMAIN}"
+    port:
+      targetPort: 8080
+    tls:
+      insecureEdgeTerminationPolicy: Redirect
+      termination: edge
+    to:
+      kind: Service
+      name: kube-ops-view
+- kind: Service
+  apiVersion: v1
+  metadata:
+    labels:
+      app: kube-ops-view
+    name: kube-ops-view
+  spec:
+    ports:
+    - port: 8080
+      protocol: TCP
+      targetPort: 8080
+    selector:
+      app: kube-ops-view
+    type: ClusterIP
+- kind: Service
+  apiVersion: v1
+  metadata:
+    labels:
+      app: kube-ops-view-redis
+    name: kube-ops-view-redis
+  spec:
+    ports:
+    - port: 6379
+      protocol: TCP
+      targetPort: 6379
+    selector:
+      app: kube-ops-view-redis
+    type: ClusterIP
+- kind: Secret
+  apiVersion: v1
+  metadata:
+    name: kube-ops-view-oauth
+  stringData:
+    authcode-client-id: "${{KUBE_OPS_VIEW_CLIENT_ID}}"
+    authcode-client-secret: "${{KUBE_OPS_VIEW_CLIENT_SECRET}}"
+parameters:
+  - name: KUBE_OPS_VIEW_APP_DOMAIN
+    description: The hostname to register for the route, leave blank for default
+    value: kube-ops-view.example.com
```

Comment:
can you make this value mandatory by removing the default (which is not going to be valid in most cases anyway). So the user will have to think about what they want here.